File: BSIGeneralAffairBO/Proposal.vb
Public Class Proposal
    Public Property ProposalToken As String
    Public Property DepartementID As Byte?
    Public Property UserID As Integer?
    Public Property VendorID As Short?
    Public Property ProposalObjective As String
    Public Property ProposalDescription As String
    Public Property ProposalRequireDate As DateTime?
    Public Property ProposalBudget As Decimal?
    Public Property ProposalNote As String
    Public Property ProposalType As String
    Public Property ProposalStatus As String
    Public Property ProposalApproveLevel As Byte?
    Public Property ProposalNegotiationNote As String
    Public Property CreatedAt As DateTime?
    Public Property UpdatedAt As DateTime?

    Public Property Department As Departement
    Public Property User As User
    Public Property Vendor As Vendor
    Public Property File As List(Of ProposalFile)
    Public Property ServiceHistory As List(Of ProposalService)

End Class

File: BSIGeneralAffairBO-C/Proposal.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace BSIGeneralAffairBO_C
{
    public class Proposal
    {
        public Proposal() { 
            this.Department = new Departement();
            this.User = new User();
            this.Employee = new Employee();
            this.Vendor = new Vendor();
            this.File = new List<ProposalFile>();
            this.ServiceHistory = new List<ProposalService>();
        }
        public string ProposalToken { get; set; }
        public byte? DepartementID { get; set; }
        public int? UserID { get; set; }
        public short? VendorID { get; set; }
        public string ProposalObjective { get; set; }
        public string ProposalDescription { get; set; }
        public DateTime? ProposalRequireDate { get; set; }
        public decimal? ProposalBudget { get; set; }
        public string ProposalNote { get; set; }
        public string ProposalType { get; set; }
        public string ProposalStatus { get; set; }
        public byte? ProposalApproveLevel { get; set; }
        public string ProposalNegotiationNote { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }

        public Departement Department { get; set; }
        public User User { get; set; }
        public Employee Employee { get; set; }
        public Vendor Vendor { get; set; }
        public List<ProposalFile> File { get; set; }
        public List<ProposalService> ServiceHistory { get; set; }
    }

}
